recipes/leveldb/all/patches/check_library_exists.1.23.patch

```diff
@@ -1,25 +1,30 @@
-diff --git a/CMakeLists.txt b/CMakeLists.txt
-index f8285b8..55258e7 100644
 --- a/CMakeLists.txt
 +++ b/CMakeLists.txt
-@@ -38,10 +38,18 @@ include(CheckIncludeFile)
+@@ -38,8 +38,12 @@ include(CheckIncludeFile)
  check_include_file("unistd.h" HAVE_UNISTD_H)
  
  include(CheckLibraryExists)
 -check_library_exists(crc32c crc32c_value "" HAVE_CRC32C)
 -check_library_exists(snappy snappy_compress "" HAVE_SNAPPY)
++if(HAVE_CRC32C)
++  find_package(Crc32c REQUIRED CONFIG)
++endif()
++if(HAVE_SNAPPY)
++  find_package(Snappy REQUIRED CONFIG)
++endif()
  check_library_exists(tcmalloc malloc "" HAVE_TCMALLOC)
  
-+find_package(Crc32c)
-+if (Crc32c_FOUND)
-+  set(HAVE_CRC32C ON)
-+endif (Crc32c_FOUND)
-+
-+find_package(Snappy)
-+if (Snappy_FOUND)
-+  set(HAVE_SNAPPY ON)
-+endif (Snappy_FOUND)
-+
  include(CheckCXXSymbolExists)
- # Using check_cxx_symbol_exists() instead of check_c_symbol_exists() because
- # we're including the header from C++, and feature detection should use the same
+@@ -268,10 +272,10 @@ if(HAVE_CLANG_THREAD_SAFETY)
+ endif(HAVE_CLANG_THREAD_SAFETY)
+ 
+ if(HAVE_CRC32C)
+-  target_link_libraries(leveldb crc32c)
++  target_link_libraries(leveldb Crc32c::crc32c)
+ endif(HAVE_CRC32C)
+ if(HAVE_SNAPPY)
+-  target_link_libraries(leveldb snappy)
++  target_link_libraries(leveldb Snappy::snappy)
```

Comment:
should we fix `cmake_target_name` in `snappy`?

Reply:
No, target name is correct in `snappy` recipe, it's `Snappy::snappy`. It's "wrong" in `leveldb` CMakeLists because they don't rely on `find_package()`, but `check_library_exists()`, so they "link" to the library directly, not the target.

Reply:
interesting. so, is there any way to transparently support such use cases, without patches?

Reply:
I don't think so, leveldb CMakeLists is fragile by design.